box2d/src/main/kotlin/ktx/box2d/worlds.kt

```diff
@@ -60,3 +61,69 @@ fun World.create(bodyDefinition: BodyDefinition): Body {
  * @see createWorld
  */
 val earthGravity = Vector2(0f, -9.8f)
+
+/**
+ * Callback lambda for ray-casts. This lambda is called when a ray-cast hits a fixture.
+ *
+ * The lambda accepts these parameters:
+ * - `[Fixture]`, the fixture hit by the ray
+ * - `[Vector2]`, the point of initial intersection
+ * - `[Vector2]`, the normal vector at the point of intersection
+ * - `[Float]`, the fraction of the distance from `start` to `end` that the intersection point is at
+ *
+ * The lambda should return:
+ * - `-1f`, ignore this fixture and continue
+ * - `0f`, terminate the ray cast
+ * - a fraction, clip the length of the ray to this point
+ * - `1f`, don't clip the ray and continue
+ *
+ * Can be used in place of [com.badlogic.gdx.physics.box2d.RayCastCallback] via Kotlin SAM conversion.
+ */
+typealias RayCastCallback = (Fixture, Vector2, Vector2, Float) -> Float
```

Comment:
I agree the global constants aren't ideal; the problem is that the returned Float has a special meaning in Box2D if it's a fraction.

For example, to get only the nearest raycast hit in the most efficient way, you have to return the fraction and then pick the nearest intersection from the multiple calls to the callback (since there is not guarantee that the hits are returned in order of nearest to furthest).

It's best explained here! http://www.iforce2d.net/b2dtut/world-querying